django_celery_beat/models.py

```diff
@@ -331,24 +372,31 @@ class Meta:
     def validate_unique(self, *args, **kwargs):
         super(PeriodicTask, self).validate_unique(*args, **kwargs)
 
-        schedule_types = ['interval', 'crontab', 'solar']
+        schedule_types = ['interval', 'crontab', 'solar', 'clocked']
         selected_schedule_types = [s for s in schedule_types
                                    if getattr(self, s)]
 
         if len(selected_schedule_types) == 0:
             raise ValidationError({
                 'interval': [
-                    'One of interval, crontab, or solar must be set.'
+                    'One of clocked, interval, crontab, or solar must be set.'
                 ]
             })
 
-        err_msg = 'Only one of interval, crontab, or solar must be set'
+        err_msg = 'Only one of clocked, interval, crontab, '\
+            'or solar must be set'
         if len(selected_schedule_types) > 1:
             error_info = {}
             for selected_schedule_type in selected_schedule_types:
                 error_info[selected_schedule_type] = [err_msg]
             raise ValidationError(error_info)
 
+        # clocked must be one off task
+        if getattr(self, "clocked"):
+            if getattr(self, "one_off", False):
```

Comment:
You could combine this to just:

    if self.clocked and self.one_off:

Reply:
Actually... given the error message there, isn't your logic wrong here and it should be 

    if self.clocked and not self.one_off:  

?

Reply:
- - it is a mistake